src/dom/Struct.ts

```diff
@@ -141,12 +141,16 @@ export class Struct extends Value(
 
   elements(): Value[] {
     let singleValueFields = Object.create(null);
-    for (const [fieldName, values] of this.allFields()) {
-      singleValueFields[fieldName] = values[values.length - 1];
+    for (const [fieldName, value] of this.fields()) {
+      singleValueFields[fieldName] = value;
     }
     return Object.values(singleValueFields);
   }
 
+  allElements(): Value[][] {
```

Comment:
This is returning an array-of-arrays, but I think we want it to have the same method signature as `elements` on line 142 above.

You can use the [`Array` method `flat()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flat) to achieve this.
```ts
return Object.values(this._fields).flat();
```